### PR TITLE
Turn on python tests

### PR DESCRIPTION
**Pull request by yzhang-23**
**PR Type**

- [ ] Breaking change
- [ ] Feature
- [X] Patch

**Brief Description**
In this PR I just turned on python test. I changed the directory name `tests/python/unit_tests/logging` to `tests/python/unit_tests/logging_new`, otherwise importing errors occur.

**Not In Scope**

**PR Checklist**

- [ ] [Is documented](https://nwchemex-project.github.io/.github/documenting/index.html)
- [ ] [Is tested](https://nwchemex-project.github.io/.github/testing/index.html)
- [ ] [Adheres to applicable organization standards](https://nwchemex-project.github.io/.github/conventions/index.html)

**TODOs**

- [ ] Task 1
- [ ] Task 2

### CODE REVIEW

**Comment by ryanmrichard** [comment]
> In this PR I just turned on python test. I changed the directory name tests/python/unit_tests/logging to tests/python/unit_tests/logging_new, otherwise importing errors occur.

I assume the problem is that there are two `logging` modules (the actual module and the one which tests the module)? If that's the case I think it's more conventional to name the testing one `test_logging` rather than `logging_new`.

**Comment by ryanmrichard** [comment]
Looking at the testing logs, this doesn't look like it's running anything other than the serial C++ unit tests. I'm specifically looking at [this](https://github.com/NWChemEx-Project/ParallelZone/actions/runs/5977077817/job/16216272029) log, the "build test release in a container" step, which shows only a single ctest test suite running.

**Comment by yzhang-23** [comment]
> Looking at the testing logs, this doesn't look like it's running anything other than the serial C++ unit tests. I'm specifically looking at [this](https://github.com/NWChemEx-Project/ParallelZone/actions/runs/5977077817/job/16216272029) log, the "build test release in a container" step, which shows only a single ctest test suite running.

My first run failed because of missing the setting of `NWX_MODULE_DIRECTORY`, then I added set `NWX_MODULE_DIRECTORY=./NWX_PyModules` in the toolchain file, which works on my local machine, but now still no python tests run, strange. Let me check more.

File: .github/Dockerfile/rebuild_base.txt
Comment (ryanmrichard) [inline .github/Dockerfile/rebuild_base.txt:1]: GitHub allows you to manually trigger workflows (although you must explicitly enable this for each workflow you want to be able to manually trigger)

**Review by ryanmrichard — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -37,7 +37,7 @@ include(nwx_versions)
 # These are the default values, used only if the user hasn't already set them.
 option(BUILD_TESTING "Should we build the tests?" OFF)
 option(BUILD_SHARED_LIBS "Should we build shared libraries?" ON)
-option(BUILD_PYBIND11_PYBINDINGS "Should we build pybind11 python bindings?" OFF)
+option(BUILD_PYBIND11_PYBINDINGS "Should we build pybind11 python bindings?" ON)
```
Comment (jwaldrop107) [inline CMakeLists.txt:40]: Do we want to change the default here, or should this be changed in the toolchain?

**Review by keceli — APPROVED** [review]
Looks good.

File: CMakeLists.txt (hunk 1 above)
Comment (keceli) [reply CMakeLists.txt:40]: I think the default should be the lean build without bindings.

Comment (ryanmrichard) [reply CMakeLists.txt:40]: I'd actually argue they should be on by default. If we're expecting users to interact with NWX through the Python bindings, they shouldn't have to enable the bindings. That said, off by default until we're ready to declare the Python bindings released is fine by me.

Comment (keceli) [reply CMakeLists.txt:40]: I think Python users would expect/prefer `pip install parallelzone` not the `cmake` build. That said, I am fine either way for the default.

File: .github/Dockerfile/rebuild_base.txt
Comment (yzhang-23) [reply .github/Dockerfile/rebuild_base.txt:1]: Manually triggered workflows do not work directly in non-default (master) branches. In addition, here rebuild_base.txt serves more like an input parameter (turn on/off some jobs in the workflow) than an event to trigger the workflow.

Comment (ryanmrichard) [reply .github/Dockerfile/rebuild_base.txt:1]: I don't follow, but that's fine. For now what matters is getting the CI switched over to Docker images.

**Comment by ryanmrichard** [comment]
🚀 [[bumpr]](https://github.com/haya14busa/action-bumpr) [Bumped!](https://github.com/NWChemEx-Project/ParallelZone/actions/runs/5979645101) **New version**:[v0.1.12](https://github.com/NWChemEx-Project/ParallelZone/releases/tag/v0.1.12) **Changes**:[v0.1.11...v0.1.12](https://github.com/NWChemEx-Project/ParallelZone/compare/v0.1.11...v0.1.12)

**Comment by yzhang-23** [comment]
@ryanmrichard @keceli Something I'm still confused about is that although I have disabled integration testing (only a placeholder for now) in my workflow, it is still running. This was not the case before I turned the python tests on. Any idea?

**Comment by keceli** [comment]
Looks like it interprets it as a regular expression where `*` means zero or more of the preceding element. So, you would need "py.*". Or, if you just have py, it would run all tests that start with py.

**Pull request merged**